07_web_endpoints/aiortc/modal_webrtc/webrtc_yolo.py

```diff
@@ -0,0 +1,420 @@
+# standard python imports...
+import os
+from pathlib import Path
+
+# ...and modal
+import modal
+
+from .modal_webrtc import ModalWebRTCPeer, ModalWebRTCServer
+
+APP_NAME = "aiortc-server-video-processing-example"
+
+# create an output volume to store the transmitted videos and model weights
+CACHE_VOLUME = modal.Volume.from_name(
+    "webrtc-yolo-cache", create_if_missing=True
+)
+CACHE_PATH = Path("/cache")
+
+assets_parent_directory = Path(__file__).parent.parent.resolve()
+
+# image
+webrtc_base_image = (
+    modal.Image.debian_slim(python_version="3.12")
+    .apt_install("python3-opencv", "ffmpeg")
+    .pip_install(
+        "fastapi[standard]==0.115.4",
+        "aiortc==1.11.0",
+        "opencv-python==4.11.0.86",
+    )
+)
+
+server_image = webrtc_base_image.add_local_dir(
+    # frontend files
+    os.path.join(assets_parent_directory, "frontend"),
+    remote_path="/frontend",
+)
+
+tester_image = webrtc_base_image.add_local_dir(
+    # video file for testing
+    os.path.join(assets_parent_directory, "media"),
+    remote_path="/media",
+)
+
+video_processing_gpu_image = (
+    modal.Image.debian_slim(python_version="3.12")
+    .apt_install("locales")
+    .run_commands(
+        # below is necessary for proper use of
+        # the onnxruntime(-gpu) package
+        # uncomment US.UTF-8 locale
+        "sed -i '/^#\\s*en_US.UTF-8 UTF-8/ s/^#//' /etc/locale.gen",
+        # generate locale
+        "locale-gen en_US.UTF-8",
+        # update locale
+        "update-locale LANG=en_US.UTF-8",
+    )
+    .apt_install("python3-opencv", "ffmpeg")
+    .env(
+        {
+            # needed to use tensorrt execution provider
+            # with onnxruntime(-gpu)
+            "LD_LIBRARY_PATH": "/usr/local/lib/python3.12/site-packages/tensorrt_libs",
+            # (maybe) needed for proper use of onnxruntime(-gpu)
+            "LANG": "en_US.UTF-8",
+        }
+    )
+    .pip_install(
+        "fastapi==0.115.12",
+        "aiortc==1.11.0",
+        "opencv-python==4.11.0.86",
+        "tensorrt==10.9.0.34",
+        "torch==2.7.0",
+        "onnxruntime-gpu==1.21.0",
+        "huggingface-hub==0.30.2",
+    )
+)
+
+# instantiate our app
+app = modal.App(APP_NAME)
+
+
+# our modal peer, a subclass of ModalWebRTCPeer
+# this class flips and incoming video stream
+# and then streams the flipped video back to the provider
+@app.cls(
+    image=video_processing_gpu_image,
```

Comment:
Worth mentioning region pinning as an option here, so you can control the latency between the client and the GPU container.